www/js/playerlib.js

```diff
@@ -1664,195 +1679,119 @@ function loadLibrary() {
-	                allAlbums.push(objCompilationAlbum);
-	                compAlbumStored = true;
-					//console.log('compilation album=' + objCompilationAlbum.album);
-				}
-			}
-			// current != prev -> lets check 
-			else {
-				// prev = last compilation album stored
-				if (allAlbumsTmp[i - 1].album == objCompilationAlbum.album) {
-					// don't store it, just reset and move on
-					objCompilationAlbum = {'album': '', 'artist': '', 'compilation': '1'};
-				}
-				// prev is a regular album, store it 
-				else {
-	                var objRegularAlbum = {'album': allAlbumsTmp[i - 1].album, 'artist': allAlbumsTmp[i - 1].artist, 'compilation': '0', 'imgurl': allAlbumsTmp[i - 1].imgurl};
-					allAlbums.push(objRegularAlbum);
-					//console.log('regular album=' + objRegularAlbum.album);
-				}
-				// last album
-				if (i == allAlbumsTmp.length - 1) {
-					// store last album
-					var objRegularAlbum = {'album': allAlbumsTmp[i].album, 'artist': allAlbumsTmp[i].artist, 'compilation': '0', 'imgurl': allAlbumsTmp[i].imgurl};
-					allAlbums.push(objRegularAlbum);
-					//console.log('regular album=' + objRegularAlbum.album);
-				}
-				// reset flag
-				compAlbumStored = false;
-			}
-		}	
+	var allArtistAlbums = Object.values(allSongs.reduce(reduceArtists, {})).reduce(function(acc, artistTracks) {
+		var artistAlbums = artistTracks.reduce(reduceAlbums, {});
+		return acc.concat(Object.values(artistAlbums));
+	}, []);
+
+	allAlbums = allArtistAlbums.map(function(albumTracks){
+		var file = albumTracks.find(function(track) { return track.file; }).file;
+		var md5 = $.md5(file.substring(0,file.lastIndexOf('/')));
+		var albumArtist = (albumTracks.find(function(track) { return track.album_artist; }) || {}).album_artist;
+		return {
+			last_modified: new Date(Math.max.apply(null, albumTracks.map(function(track){ return new Date(track.last_modified); }))),
+			album: albumTracks.find(function(track) { return track.album; }).album,
+			artist: albumArtist || albumTracks.find(function(track) { return track.artist; }).artist,
+			imgurl: '/imagesw/thmcache/' + encodeURIComponent(md5) + '.jpg'
+		};
+	});
+
+	allAlbumCovers = allAlbums.slice();
+
+	// Filter album from songs after generating allAlbums && allAlbumCovers
+	var albumFilter = LIB.filters.albums;
+	if (albumFilter.length) {
+		allSongs = allSongs.filter(function(track) {
+			return albumFilter.includes(keyAlbum(track));
+		});
 	}
-	// only one album in list
-	else if (allAlbumsTmp.length == 1) {
-		var objRegularAlbum = {'album': allAlbumsTmp[0].album, 'artist': allAlbumsTmp[0].artist, 'compilation': '0', 'imgurl': allAlbumsTmp[0].imgurl}; // store the one and only album
-		allAlbums.push(objRegularAlbum);
+
+	// sort Genres, Artists, Albums
+	allGenres.sort();
+	allAlbumCovers.sort(function(a, b) {
+		return b.last_modified - a.last_modified;
```

Comment:
@moodeaudio We don't need to force this change on everyone. This is just my personal preference for sorting (newest first)